Include/Rover/LinearRegressionModel.hpp

```diff
@@ -0,0 +1,73 @@
+#ifndef ROVER_LINEAR_REGRESSION_MODEL_HPP
+#define ROVER_LINEAR_REGRESSION_MODEL_HPP
+#include <dlib/matrix.h>
+
+namespace Rover {
+  //! Models a trial using linear regression.
+  /*!
+    \tparam T The Trial to model.
+  */
+  template<typename T>
+  class LinearRegressionModel {
+    public:
+
+      //! The Trial to model.
+      using Trial = T;
+
+      //! The result of the trial.
+      using Result = typename Trial::Sample::Result;
+
+      //! Constructs a linear regression model for a trial.
+      /*!
+        \param trial The trial to model.
+      */
+      LinearRegressionModel(const Trial& trial);
+
+      template<typename... Arg>
+      Result operator ()(Arg&&... arg) const;
+
+    private:
+      static dlib::matrix<Result> compute_transformation_vector(const T& trial);
+
+    private:
+      dlib::matrix<Result> m_transformation;
+  };
+
+  template<typename T>
+  LinearRegressionModel<T>::LinearRegressionModel(const T& trial)
+    : m_transformation(compute_transformation_vector(trial)) {}
+
+  template<typename T>
+  template<typename... Arg>
+  typename LinearRegressionModel<T>::Result
+      LinearRegressionModel<T>::operator ()(Arg&&... arg) const {
+    auto x = dlib::matrix<Result, 1, std::tuple_size_v<
+      typename Trial::Sample::Parameters> + 1>();
+    x = { 1., std::forward<Arg>(arg)... };
```

Comment:
C++ ```initialization_list``` does not support forwarding and generally its use should be avoided, it's a huge blunder. Can you initialize this another way?